src/gridcharger/huawei/Controller.cpp

```diff
@@ -37,14 +37,24 @@ void Controller::init(Scheduler& scheduler)
 
 void Controller::enableOutput()
 {
+    if (_oOutputEnabled.value_or(false)) { return; }
+
+    _setProduction(true);
     if (_huaweiPower <= GPIO_NUM_NC) { return; }
     digitalWrite(_huaweiPower, 0);
+
+    _oOutputEnabled = true;
 }
 
 void Controller::disableOutput()
 {
+    if (!_oOutputEnabled.value_or(true)) { return; }
+
+    _setProduction(false);
     if (_huaweiPower <= GPIO_NUM_NC) { return; }
     digitalWrite(_huaweiPower, 1);
+
+    _oOutputEnabled = false;
```

Comment:
I think that we are currently spamming the PSU with can messages to enable or disable the output when the pin `_huaweiPower` is not in use. `_oOutputEnabled` is only set to true or false when the power pin is set.

```suggestion
    _oOutputEnabled = false;
    
    if (_huaweiPower <= GPIO_NUM_NC) { return; }
    digitalWrite(_huaweiPower, 1);
```